Tests/GRPCInterceptorsTests/PeerAddressTests.swift

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2025, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import GRPCInterceptors
+import Testing
+
+@Suite("PeerAddress tests")
+struct PeerAddressTests {
+  @Test("IPv4 addresses are correctly parsed")
+  func testIPv4() {
+    let address = PeerAddress("ipv4:10.1.2.80:567")
+    #expect(address == .ipv4(address: "10.1.2.80", port: 567))
+  }
+
+  @Test("IPv6 addresses are correctly parsed")
+  func testIPv6() {
+    let address = PeerAddress("ipv6:2001::130F:::09C0:876A:130B:1234")
+    #expect(address == .ipv6(address: "2001::130F:::09C0:876A:130B", port: 1234))
+  }
+
+  @Test("Unix domain sockets are correctly parsed")
+  func testUDS() {
+    let address = PeerAddress("unix:some-path")
+    #expect(address == .unixDomainSocket(path: "some-path"))
+  }
+
+  @Test("Unrecognised addresses are correctly parsed")
+  func testOther() {
+    let address = PeerAddress("in-process:1234")
+    #expect(address == .other("in-process:1234"))
+  }
```

Comment:
I'm still not sure whether we should include addresses we can't parse. I _think_ my lean is to not include them because we're then more likely to catch parsing bugs (I think a user is more likely to report an issue if the information is missing rather than not quite appearing as it should).

Reply:
Hm, I'm not sure about this. I feel like silently skipping information is worse than showing it in a not-ideal way.

Reply:
I'm not sure I agree: we don't know what the value is so we should treat it with a bit of suspicion.

Reply:
I can think of a few things:
- If, for some reason, the string is very dynamic (i.e. changes for each RPC) then we could pollute the users telemetry collection service with thousands of potentially useless values.
- A very long could cause issues if the collector doesn't handle long strings well.

In practice these both seem pretty unlikely, but they are possible. 

IMO the more compelling reason to not set it is that users are more likely to complain if the value is missing and then we can figure out why parsing failed so we can set it correctly.

I also find it weird that we can return `.other` when we know it's ipv4/ipv6 but we can't parse it, because it doesn't meet our expectations of what that address type should be.